crates/fuel-core/src/query/balance/asset_query.rs

```diff
@@ -99,12 +102,25 @@ impl<'a> AssetsQuery<'a> {
                     } else {
                         return Err(anyhow::anyhow!("The coin is not UTXO").into());
                     };
-                    // TODO: Fetch coin in a separate thread
-                    let coin = self.database.coin(id)?;
-
-                    Ok(CoinType::Coin(coin))
+                    Ok(id)
                 })
+            });
+
+        futures::stream::StreamExt::chunks(stream, database.batch_size)
```

Comment:
We could just refer to `self.database` here
```suggestion
        futures::stream::StreamExt::chunks(stream, self.database.batch_size)
```

Reply:
Or add a getter for `batch_size` :)